src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -2694,6 +2642,233 @@ def get_enable_azure_rbac(self, enable_validation: bool = False, **kwargs) -> bo
 
         return self._get_enable_azure_rbac(enable_validation=True)
 
+    def get_api_server_authorized_ip_ranges(self) -> List[str]:
+        """Obtain the value of api_server_authorized_ip_ranges.
+
+        This function supports the option of enable_validation. When enabled, it will check if load_balancer_sku equals
+        to "basic" when api_server_authorized_ip_ranges is assigned, if so, raise the MutuallyExclusiveArgumentError.
+        This function will normalize the parameter by default. It will split the string into a list with "," as the
+        delimiter.
+
+        :return: empty list or list of strings
+        """
+        # read the original value passed by the command
+        api_server_authorized_ip_ranges = self.raw_param.get(
+            "api_server_authorized_ip_ranges"
+        )
+        # try to read the property value corresponding to the parameter from the `mc` object
+        read_from_mc = False
+        if (
+            self.mc and
+            self.mc.api_server_access_profile and
+            self.mc.api_server_access_profile.authorized_ip_ranges is not None
+        ):
+            api_server_authorized_ip_ranges = (
+                self.mc.api_server_access_profile.authorized_ip_ranges
+            )
+            read_from_mc = True
+
+        # normalize
+        if not read_from_mc:
+            api_server_authorized_ip_ranges = [
+                x.strip()
+                for x in (
+                    api_server_authorized_ip_ranges.split(",")
+                    if api_server_authorized_ip_ranges
+                    else []
+                )
+            ]
+
+        # validation
+        if api_server_authorized_ip_ranges:
+            if self._get_load_balancer_sku(enable_validation=False) == "basic":
```

Comment:
use ignore case equal?

Reply:
Updated in commit [#62fd075](https://github.com/Azure/azure-cli/pull/19539/commits/62fd075d4d50739e6b1e47fa81de0ed79dcdf962).